tests/integration/aof-multi-part.tcl

```diff
@@ -1166,5 +1166,138 @@ tags {"external:skip"} {
             set d2 [r debug digest]
             assert {$d1 eq $d2}
         }
+
+        test "AOF will not open incr file until the first AOFRW success when AOF is dynamically enabled" {
+            # Clear all data and delete appenddir to get an empty redis
+            r flushall
+            r config set appendonly no
+            catch {exec rm -rf $aof_dirpath}
+            r debug loadaof
+
+            assert_equal 0 [r dbsize]
+
+            # Increase AOFRW execution time to give us enough time to kill it
+            r config set rdb-key-save-delay 10000000
+
+            # Start write load
+            set load_handle0 [start_write_load $master_host $master_port 10]
+
+            wait_for_condition 50 100 {
+                [r dbsize] > 0
+            } else {
+                fail "No write load detected."
+            }
+
+            r config set appendonly yes
+            
+            # Make sure AOFRW was scheduled 5 times
+            set total_forks [s total_forks]
+            set maxtries 1000
+            set delay 10
+            while {[incr maxtries -1] >= 0} {
+                if {[s total_forks] == [expr $total_forks + 5]} {
+                    break
+                }
+                catch {exec kill -9 [get_child_pid 0]}
```

Comment:
maybe we should wait to see that there is a child, or check the value returned by get_child_pid before doing the kill?